### PR TITLE
send events

### PR DESCRIPTION
**Pull request by n3wscott**

### CODE REVIEW

File: test/e2e/cmd/producer/main.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"github.com/kelseyhightower/envconfig"
```
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"log"
+	"os"
+	"time"
```
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/config/recorder/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+metadata:
+  name: event-watcher-{{ .namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher-{{ .namespace }}
+subjects:
+  - kind: ServiceAccount
+    name: recorder
+    namespace: {{ .namespace }}
```
Comment (mattmoor) [inline test/e2e/config/recorder/rbac.yaml:50]: Add trailing newline:
```suggestion
    namespace: {{ .namespace }}

```

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/config/recorder/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (mattmoor) [inline test/e2e/config/recorder/rbac.yaml:60]: Add trailing newline:
```suggestion
    namespace: {{ .namespace }}

```

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/config/recorder/rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (mattmoor) [inline test/e2e/config/recorder/rbac.yaml:60]: Add trailing newline:
```suggestion
    namespace: {{ .namespace }}

```

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

File: test/e2e/direct_test.go [hunk 6]
```diff
@@ -17,16 +17,28 @@ limitations under the License.
 package rabbit_test
 
 import (
+	"fmt"
+	"knative.dev/eventing/pkg/test/observer"
+	"knative.dev/pkg/injection/sharedmain"
```
Comment (mattmoor) [inline test/e2e/direct_test.go:22]: Format Go code:
```suggestion

```

File: test/e2e/direct_test.go [hunk 7]
```diff
@@ -17,16 +17,28 @@ limitations under the License.
...
 	"strings"
 	"testing"
 	"time"
```
Comment (mattmoor) [inline test/e2e/direct_test.go:26]: Format Go code:
```suggestion

	"knative.dev/eventing/pkg/test/observer"
	"knative.dev/pkg/injection/sharedmain"

```

File: test/e2e/direct_test.go [hunk 8]
```diff
@@ -17,16 +17,28 @@ limitations under the License.
...
 
 	"github.com/n3wscott/rigging"
+	"knative.dev/eventing/pkg/test/observer/recorder-collector"
```
Comment (mattmoor) [inline test/e2e/direct_test.go:28]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

File: test/e2e/direct_test.go (hunk 6 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:22]: Format Go code:
```suggestion

```

File: test/e2e/direct_test.go (hunk 7 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:26]: Format Go code:
```suggestion

	"knative.dev/eventing/pkg/test/observer"
	"knative.dev/pkg/injection/sharedmain"

```

File: test/e2e/direct_test.go (hunk 8 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:28]: Format Go code:
```suggestion
	recorder_collector "knative.dev/eventing/pkg/test/observer/recorder-collector"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 1 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 2 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

File: test/e2e/direct_test.go (hunk 6 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:22]: Format Go code:
```suggestion

```

File: test/e2e/direct_test.go (hunk 7 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:26]: Format Go code:
```suggestion

	"knative.dev/eventing/pkg/test/observer"
	"knative.dev/pkg/injection/sharedmain"

```

File: test/e2e/direct_test.go (hunk 8 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:28]: Format Go code:
```suggestion
	recorder_collector "knative.dev/eventing/pkg/test/observer/recorder-collector"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"github.com/kelseyhightower/envconfig"
```
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+	"log"
+	"os"
+	"time"
```
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

File: test/e2e/direct_test.go (hunk 6 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:22]: Format Go code:
```suggestion

```

File: test/e2e/direct_test.go (hunk 7 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:26]: Format Go code:
```suggestion

	"knative.dev/eventing/pkg/test/observer"
	"knative.dev/pkg/injection/sharedmain"

```

File: test/e2e/direct_test.go (hunk 8 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:28]: Format Go code:
```suggestion
	recorder_collector "knative.dev/eventing/pkg/test/observer/recorder-collector"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 9 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 10 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

File: test/e2e/direct_test.go (hunk 6 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:22]: Format Go code:
```suggestion

```

File: test/e2e/direct_test.go (hunk 7 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:26]: Format Go code:
```suggestion

	"knative.dev/eventing/pkg/test/observer"
	"knative.dev/pkg/injection/sharedmain"

```

File: test/e2e/direct_test.go (hunk 8 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:28]: Format Go code:
```suggestion
	recorder_collector "knative.dev/eventing/pkg/test/observer/recorder-collector"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/cmd/producer/main.go (hunk 9 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:21]: Format Go code:
```suggestion

```

File: test/e2e/cmd/producer/main.go (hunk 10 above)
Comment (mattmoor) [inline test/e2e/cmd/producer/main.go:24]: Format Go code:
```suggestion
	"time"

	"github.com/kelseyhightower/envconfig"
```

File: test/e2e/direct_test.go (hunk 6 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:22]: Format Go code:
```suggestion

```

File: test/e2e/direct_test.go (hunk 7 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:26]: Format Go code:
```suggestion

	"knative.dev/eventing/pkg/test/observer"
	"knative.dev/pkg/injection/sharedmain"

```

File: test/e2e/direct_test.go (hunk 8 above)
Comment (mattmoor) [inline test/e2e/direct_test.go:28]: Format Go code:
```suggestion
	recorder_collector "knative.dev/eventing/pkg/test/observer/recorder-collector"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/main_test.go [hunk 11]
```diff
@@ -19,12 +19,16 @@ limitations under the License.
 package rabbit_test
 
 import (
+	"context"
 	"fmt"
+	"github.com/n3wscott/rigging/pkg/lifecycle"
```
Comment (mattmoor) [inline test/e2e/main_test.go:24]: Format Go code:
```suggestion

```

File: test/e2e/main_test.go [hunk 12]
```diff
@@ -19,12 +19,16 @@ limitations under the License.
...
 	"os"
 	"testing"
 	"text/template"
```
Comment (mattmoor) [inline test/e2e/main_test.go:27]: Format Go code:
```suggestion
	"text/template"

	"github.com/n3wscott/rigging/pkg/lifecycle"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: test/e2e/main_test.go (hunk 11 above)
Comment (mattmoor) [inline test/e2e/main_test.go:24]: Format Go code:
```suggestion

```

File: test/e2e/main_test.go (hunk 12 above)
Comment (mattmoor) [inline test/e2e/main_test.go:27]: Format Go code:
```suggestion
	"text/template"

	"github.com/n3wscott/rigging/pkg/lifecycle"
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

**Comment by vaikas** [comment]
Besides needing rebase, is there something wrong with the setup for the secrets?

```
2020/10/05 18:34:42 smoke-broker-trigger-hbavypkb is not ready, ExchangeCredentialsUnavailable: Failed to get arguments for creating exchange: rabbit "rigging-oxwsdpwp/rabbitmqc" not ready
```

**Comment by n3wscott** [comment]
> Besides needing rebase, is there something wrong with the setup for the secrets?
> 
> ```
> 2020/10/05 18:34:42 smoke-broker-trigger-hbavypkb is not ready, ExchangeCredentialsUnavailable: Failed to get arguments for creating exchange: rabbit "rigging-oxwsdpwp/rabbitmqc" not ready
> ```

Did you run  `./test/kind/prerequisites.sh`?

See  ./test/kind/README.md

**Review by mattmoor — COMMENTED** [review]
/lgtm

**Pull request merged**